src/main/java/com/conveyal/r5/profile/FastRaptorWorker.java

```diff
@@ -323,74 +348,61 @@ private void advanceScheduledSearchToPreviousMinute (int nextMinuteDepartureTime
                 doScheduledSearchForRound(scheduleState[round - 1], scheduleState[round]);
                 raptorTimer.scheduledSearchTransit.stop();
 
-                // If there are frequency routes, we will be randomizing the offsets of those routes.
+                // If there are frequency routes, we will be randomizing the schedules of those routes.
                 // First perform a frequency search using worst-case boarding time to provide a tighter upper bound on
-                // total travel time. Each randomized schedule will improve on these travel times.
-                // This is only helpful for Monte Carlo, not half-headway. Even in Monte Carlo mode,
-                // perhaps we should only do it when iterationsPerMinute is high (2 or more?).
-                if (transit.hasFrequencies && boardingMode == MONTE_CARLO) {
+                // total travel time. Each randomized schedule will then improve on these travel times.
+                // This should be a pure optimization, which is only helpful for Monte Carlo mode (not half-headway).
+                // Perhaps we should only do it when iterationsPerMinute is high (2 or more?)
+                if (transit.hasFrequencies && boardingMode == MONTE_CARLO && ENABLE_OPTIMIZATIONS) {
```

Comment:
Should this optimization flag be more specific? `ENABLE_OPTIMIZATION_X`

I can imagine a "benchmark" suite that would enable/disable them individually to compare.

Reply:
Agreed, I will split out feature flags for each separate optimization, adding one for the core optimization we discussed today: filtering the routes to explore at all.